mdaviz/mda_file_table_view.py

```diff
@@ -0,0 +1,179 @@
+"""
+Select data fields for plotting: QTableView.
+
+Uses :class:`mda_file_table_model.MDAFileTableModel`.
+
+.. autosummary::
+
+    MDAFileTableView
+"""
+
+from PyQt5 import QtCore
+from PyQt5 import QtWidgets
+
+
+from . import utils
+from .mda_file_table_model import ColumnDataType
+from .mda_file_table_model import FieldRuleType
+from .mda_file_table_model import TableColumn
+from .mda_file_table_model import TableField
+
+HEADERS = "Field", "X", "Y", "Mon", "Norm", "PV", "DESC", "Unit"
+
+COLUMNS = [
+    TableColumn("Field", ColumnDataType.text),
+    TableColumn("X", ColumnDataType.checkbox, rule=FieldRuleType.unique),
+    TableColumn("Y", ColumnDataType.checkbox, rule=FieldRuleType.multiple),
+    TableColumn("Mon", ColumnDataType.checkbox, rule=FieldRuleType.unique),
+    TableColumn("Norm", ColumnDataType.checkbox, rule=FieldRuleType.multiple),
+    TableColumn("PV", ColumnDataType.text),
+    TableColumn("DESC", ColumnDataType.text),
+    TableColumn("Unit", ColumnDataType.text),
+]
+
+
+class MDAFileTableView(QtWidgets.QWidget):
+    ui_file = utils.getUiFileName(__file__)
+
+    def __init__(self, parent):
+        """
+        Create the table view and connect with its parent.
+
+        PARAMETERS
+
+        parent object:
+            Instance of mdaviz.mda_file.MDAFile
+        """
+
+        self.mda_file = parent
+        super().__init__()
+        utils.myLoadUi(self.ui_file, baseinstance=self)
+        self.setup()
+
+    def setup(self):
+        self.setData()
+        # Configure the horizontal header to resize based on content.
+        header = self.tableView.horizontalHeader()
+        header.setSectionResizeMode(QtWidgets.QHeaderView.ResizeToContents)
+
+    def data(self):
+        """Return the data from the table view:
+              self.data= fileInfo, fields
+        with:
```

Comment:
`with:`?

Reply:
This documentation seems to be out of place.

Reply:
moved to setData